bindings/kvp/src/main/java/org/n52/sos/decode/kvp/AbstractKvpDecoder.java

```diff
@@ -355,23 +357,43 @@ private TemporalFilter createTemporalFilterFromValue(String value, String valueR
             temporalFilter.setOperator(TimeOperator.TM_Equals);
             temporalFilter.setTime(ti);
         } else if (times.length == 2) {
-            DateTime start = DateTimeHelper.parseIsoString2DateTime(times[0]);
-            // check if end time is a full ISO 8106 string
-            int timeLength = DateTimeHelper.getTimeLengthBeforeTimeZone(times[1]);
-            DateTime origEnd = DateTimeHelper.parseIsoString2DateTime(times[1]);
-            DateTime end = DateTimeHelper.setDateTime2EndOfMostPreciseUnit4RequestedEndPosition(origEnd, timeLength);
+            LOGGER.debug("Parsing temporal filter, start: {}, end: {}", times[0], times[1]);
+            DateTime start = null;
+            DateTime end = null;
+            if (times[0].startsWith(ISO_8601_DURATION_INDICATOR)) {
+                Period periodBeforeEndTime = Period.parse(times[0]);
+                // check if end time is a full ISO 8106 string
+                int timeLength = DateTimeHelper.getTimeLengthBeforeTimeZone(times[1]);
+                DateTime origEnd = DateTimeHelper.parseIsoString2DateTime(times[1]);
+                end = DateTimeHelper.setDateTime2EndOfMostPreciseUnit4RequestedEndPosition(origEnd, timeLength);
+                start = origEnd.minus(periodBeforeEndTime);
+
+            } else if (times[1].startsWith(ISO_8601_DURATION_INDICATOR)) {
+                start = DateTimeHelper.parseIsoString2DateTime(times[0]);
+                Period periodAfterStartTime = Period.parse(times[1]);
+                end = start.plus(periodAfterStartTime);
+            } else {
+                start = DateTimeHelper.parseIsoString2DateTime(times[0]);
+                end = parseEndTime(times[1]);
```

Comment:
I would have thought that this case would be identical to the original case?  Do you not have to adjust the end time?

 DateTime origEnd = DateTimeHelper.parseIsoString2DateTime(times[1]);
+                end = DateTimeHelper.setDateTime2EndOfMostPreciseUnit4RequestedEndPosition(origEnd, timeLength);